### DISCUSSION TITLE
getting Unknown field `expireDate` for select statement on model ActivationCode. error

### DISCUSSION THREAD

**Discussion by pishguy** [prisma/prisma#19989] [Q&A] [answered]
After installing `Prisma` in `NestJs` i want to select some table columns such as `code` and `expireDate` in this `Prisma` query:

```js
const activationCodes = await this.prisma.activationCode.findFirst({
  where: {
    profileId: user.profile.id,
    code:dto.code
  },
  select:{
    code:true,
    expireDate:true,
  },
});
```

for this query i'm getting this error:

```
 Type '{ code: true; expireDate: true; }' is not assignable to type 'ActivationCodeSelect'.
Object literal may only specify known properties, and 'expireDate' does not exist in type 'ActivationCodeSelect'.
```

and when i made specific type of `select` like with:

```js
import { Prisma } from "@prisma/client"

export const activationCodeSelect = {
    code: true,
    expireDate: true,
}

export type ActivationCodeSelect = Prisma.ActivationCodeGetPayload<{ select: typeof activationCodeSelect }>
```

i get this error:

```
Unknown field `expireDate` for select statement on model ActivationCode. Available options are listed in green.
```

updated query:

```js
import { activationCodeSelect } from 'prisma/types/activationCodeType';

const activationCodes = await this.prisma.activationCode.findFirst({
  where: {
    profileId: user.profile.id,
    code:dto.code
  },
  select:activationCodeSelect,
});
```

how can i solve this issue?

`prisma` model:

```prisma
model ActivationCode {
  id         Int      @id @default(autoincrement())
  profileId  Int?
  code       Int
  profile    Profile? @relation("ActivationCodesToProfile", fields: [profileId], references: [id])
  used  Boolean @default(false)
  expireDate DateTime @default(now())

  @@index([profileId], map: "ActivationCode_profileId_fkey")
}
```

`index.d.ts` from `node_modules/.pnpm/@prisma+client@4.15.0_prisma@4.16.1/node_modules/.prisma/client/index.d.ts` :

```js
export type ActivationCodeSelect = {
  id?: boolean
  profileId?: boolean
  code?: boolean
  used?: boolean
  expireDate?: boolean
  profile?: boolean | ProfileArgs
}
```

**Comment by nurul3101** [comment]
Hello @pishguy 👋 

Can you confirm that the error still persists even after invoking `npx prisma generate` command?

Your Prisma Client may be outdated or not properly regenerated after updating the schema. Running `prisma generate` to regenerate the Prisma Client with the latest schema should ensure that schema is in sync with Prisma Client. This should update the types, including ActivationCodeSelect.

**Reply by nurul3101** [reply]
@pishguy I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by pishguy** [reply, marked as answer]
@nurul3101 problem solved after restarting project Docker container :(

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know 🙌
